pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -346,11 +362,14 @@ func foreachServer(client *gophercloud.ServiceClient, opts servers.ListOptsBuild
 	return err
 }
 
-func getServerByName(client *gophercloud.ServiceClient, name types.NodeName) (*servers.Server, error) {
+func getServerByName(client *gophercloud.ServiceClient, name types.NodeName, showOnlyActive bool) (*servers.Server, error) {
 	opts := servers.ListOpts{
-		Name:   fmt.Sprintf("^%s$", regexp.QuoteMeta(mapNodeNameToServerName(name))),
-		Status: "ACTIVE",
+		Name: fmt.Sprintf("^%s$", regexp.QuoteMeta(mapNodeNameToServerName(name))),
 	}
+	if showOnlyActive {
```

Comment:
In the interested of being backward compatible and keeping the diff small, `showOnlyActive` should default to true.  It is tricky to default a go boolean variable to true, so may be we can flip the variable name - `showAll`. What you think?

Reply:
On second thoughts, it doesn't matter much. so I am going to lgtm.